crates/server/Cargo.toml

```diff
@@ -48,7 +48,7 @@ dnssec-ring = [
     "hickory-proto/dnssec-ring",
     "hickory-resolver/dnssec-ring",
 ]
-dnssec = []
+dnssec = ["hickory-recursor?/dnssec"]
```

Comment:
I'm unfamiliar with this `?` usage in features, what does it do?

Reply:
it *avoids* enabling the optional dependency `hickory-recursor` when the `hickory-server.dnssec` feature is enabled but the `hickory-server.recursor` feature was not.

in other words, `hickory-server.dnssec` enables the `hickory-recursor.dnssec` *if and only if* the opt-in `hickory-recursor` dependency was already enabled by other feature (in this case, that other feature is `hickory-server.recursor`)

More details in https://doc.rust-lang.org/cargo/reference/features.html#dependency-features